src/org/javarosa/core/model/FormDef.java

```diff
@@ -1283,22 +1283,16 @@ public void collapseIndex(FormIndex index, List<Integer> indexes, List<Integer>
 
     public FormIndex buildIndex(List<Integer> indexes, List<Integer> multiplicities, List<IFormElement> elements) {
         FormIndex cur = null;
-        List<Integer> curMultiplicities = new ArrayList<Integer>();
+        List<Integer> curMultiplicities = new ArrayList<>();
         curMultiplicities.addAll(multiplicities);
 
-        List<IFormElement> curElements = new ArrayList<IFormElement>();
+        List<IFormElement> curElements = new ArrayList<>();
         curElements.addAll(elements);
 
         for (int i = indexes.size() - 1; i >= 0; i--) {
             int ix = indexes.get(i);
             int mult = multiplicities.get(i);
 
-            // ----begin unclear why this is here... side effects???
```

Comment:
I did a quick double check that these all do appear like simple getters with no side effects.